elooplib/eventloop.cpp

```diff
@@ -63,37 +62,76 @@ int EventLoop::handleConnection(std::shared_ptr<Node> &hst)
     return -1;
 }
 
+EventLoop::nodePointerType EventLoop::waitersSearch(int fd)
+{
+    auto nodeComp = [&fd](nodeType &n)
+    {
+        return fd == n.second->getFd();
+    };
+    auto wp = find_if(begin(wait_q), end(wait_q), nodeComp);
+    return wp;
+}
+
+void EventLoop::blockClinet(nodePointerType &node_p)
+{
+    wait_q.push_back(std::move(*node_p));
+    node_p = nodes.erase(node_p);
+    --node_p;
+}
+
 void EventLoop::run()
 {
-    //auto thread_pool = ThreadPool::getInstance(); TODO
     while (!stop)
     {
-        
+
         int rc = poll(fds.data(), max_pollfd_pos, 0);
         if (rc < 0)
             throw serverExcept("poll()");
 
-        for (nodeRetType node_p = nodes.begin(); node_p != nodes.end(); node_p++)
+        for (nodePointerType node_p = nodes.begin(); node_p != nodes.end(); node_p++)
         {
-            auto &[pos, node] = *node_p;
+            auto [pos, node] = *node_p;
             if (fds[pos].revents == POLLIN)
             {
                 int data = handleConnection(node);
```

Comment:
It's not data, it's `needsThreadPool`

Reply:
Please make every return type meaningful. Introuduce type aliases. E.g.: 
`using fd_t = int; using error_t = int`

Reply:
Also rename variables to appropriate names, code is unreadable